File: packages/perspective/test/js/test_node.js
/******************************************************************************
 *
 * Copyright (c) 2017, the Perspective Authors.
 *
 * This file is part of the Perspective library, distributed under the terms of
 * the Apache License 2.0.  The full license can be found in the LICENSE file.
 *
 */

const perspective = require('../../src/js/perspective.js');
const load_asmjs = require("../../build/asmjs/psp.js").load_perspective;
const fs = require('fs');

const RUNTIMES = {
    ASMJS: perspective(load_asmjs({
        wasmJSMethod: "asmjs",
        memoryInitializerPrefixURL: 'build/asmjs/',
        asmjsCodeFile: "asmjs/psp.js",
        ENVIRONMENT: "NODE"
    })),
    WASM: {}
}

if (typeof WebAssembly !== "undefined") {
    const load_wasm = require("../../build/wasm_sync/psp.js").load_perspective;
    const wasm = fs.readFileSync('./build/wasm_sync/psp.wasm');
    RUNTIMES["WASM"] = perspective(load_wasm({
        wasmBinary: wasm,
        wasmJSMethod: 'native-wasm',
        ENVIRONMENT: "NODE"
    }));
}

const constructor_tests = require("./constructors.js");
const pivot_tests = require("./pivots.js");
const update_tests = require("./updates.js");
const filter_tests = require("./filters.js");

describe("perspective.js", function() {

    Object.keys(RUNTIMES).forEach(function(mode) {

        (typeof WebAssembly === 'undefined' && mode === "WASM" ? xdescribe : describe)(mode, function() {
    
            constructor_tests(RUNTIMES[mode]);
            pivot_tests(RUNTIMES[mode]);
            update_tests(RUNTIMES[mode]);
            filter_tests(RUNTIMES[mode]);

        });

    });

});

File: python/node/perspective/tests/node/test_node.py
# *****************************************************************************
#
# Copyright (c) 2019, the Perspective Authors.
#
# This file is part of the Perspective library, distributed under the terms of
# the Apache License 2.0.  The full license can be found in the LICENSE file.
#

from perspective.node import Perspective


class TestNode(object):
    def test_table(self):
        psp = Perspective()
        psp.start()
        psp.table({'x': 'integer', 'y': 'string', 'z': 'float'}, {'index': 'x'})

        psp.update([{'x': 1, 'y': 'a', 'z': 1.0}])
        psp.update([{'x': 2, 'y': 'b', 'z': 2.0}])
        psp.update([{'x': 3, 'y': 'c', 'z': 3.0}])
        psp.update([{'x': 4, 'y': 'd', 'z': 4.0}])
        psp.update([{'x': 5, 'y': 'test', 'z': 4.5}])
        psp.update([{'x': 7, 'y': 'test2', 'z': 2.4}])
        # psp.remove([1, 2])
        print(psp.to_json())

        v = psp.view({'columns': ['x', 'y', 'z'],
                      'aggregates': {'x': 'sum'},
                      'row_pivots': ['x'],
                      'filter': [['y', 'in', ['test']]]})

        print(v.to_json())

        psp.stop()
